kuiper-game/Domain/Ship/MiningScan.cs

```diff
@@ -27,17 +28,15 @@ public string StartEvent()
 
         public string EndEvent()
         {
-            if(DiceFactory.Roller().D100(chance,0))
-            {
-                var asteroid = CelestialBodies.Asteroid.Create("Scanned Asteroid",3.2, SolarSystemLocator.SolarSystem.SolarSystemService.GetBody("Sun"),CelestialBodies.CelestialBodyType.Asteroid );
-                var distance = SolarSystemLocator.SolarSystem.SolarSystemService.GetDistanceInAu(
-                    SolarSystemLocator.SolarSystem.Captain.Ship.CurrentLocation,
-                    asteroid
-                );
-                return $"Asteroid scanning complete. Found a candidate {distance}km from {SolarSystemLocator.SolarSystem.Captain.Ship.CurrentLocation}. Set a course to begin mining the asteroid.";
-            }
-            return $"Asteroid scanning complete, but found no asteroids with a yield within the scanners parameters.";
-
+            if (!DiceFactory.Roller().D100(chance, 0))
+                return $"Asteroid scanning complete, but found no asteroids with a yield within the scanners parameters.";
+            
+            var asteroid = CelestialBody.Create<Asteroid>("Scanned Asteroid",3.2, SolarSystemLocator.SolarSystem.SolarSystemService.GetBody("Sun"));
+            var distance = SolarSystemLocator.SolarSystem.SolarSystemService.GetDistanceInAu(
```

Comment:
Oof.. this API is not looking too hot.
Locator.Instance.Instance.method.

Anyway to improve on this through DI?

Reply:
Im actually surprised there atleast isnt a helper.

SolarSystemLocator.SolarSystem.SolarSystemService.GetDistanceInAu
Is using SolarSystemLocator.SolarSystem.Captain.Ship.CurrentLocation

Which means that SolarSystemService is depending on SolarSystem (although the API would lead you to think otherwise, because solarsystemservice is a property of SolarSystem.

Reply:
Heh - yep, but fixing that was outside the scope of this PR 😅

But I agree; we should definitely address proper use of IoC principles (and dependency chains) sometime